Artur

artekm opened the pull request.

artekm committed:
-updated ScheduleGenerator, working with last day shorter
-new PropertiesReader, working with external properties file
-new ExcelCreator
-temporary solution to write the generated workbook

artekm committed:
A bit better loop for ScheduleGenerator

artekm committed:
Small refactoring of ScheduleGenerator

olegmiercoles commented on application.properties:
```diff
@@ -0,0 +1,6 @@
+application.info=demo sources for work with files presentation
+java.version=1.8
+
+excel.defaultName=excelFileFromLocationB
```
Usually we don't want to keep something like excelFromA/B. Just proper working defaults.

artekm replied:
These names were for my own temporary use - I wanted to know if the defaults were read from correct location. Now I changed them to just "schedule", but left "scheduleJAR" for a while. If you want to make sure that it is read from JAR, just build executable JAR and run the program without application.parameters files. Your  generated schedule will get name scheduleJAR.xlsx
But I must remember to change that name when the code is finished.

olegmiercoles commented on config/application.properties:
```diff
@@ -0,0 +1,6 @@
+application.info=demo sources for work with files presentation
+java.version=1.8
+
+excel.defaultName=excelFileFromLocationA
```
Usually we don't want to keep something like excelFromA/B. Just proper working defaults.

olegmiercoles commented on config/application.properties:
```diff
@@ -0,0 +1,6 @@
+java.version=1.8
+
+excel.defaultName=excelFileFromLocationA
+excel.defaultExtension=xls
```
For now probably we rid off from keeping both extensions xlx and xlsx, as well as both dependencies. Let's keep to newer one - xlsx.

artekm replied:
OK

olegmiercoles commented on src/main/java/pl/itacademy/schedule/ScheduleGeneratorApp.java:
```diff
@@ -1,29 +1,70 @@
+		ParametersReader.UsagePrinter usagePrinter = new ParametersReader.UsagePrinter();
+		ParametersReader parametersReader = new ParametersReader();
+		EnteredParameters enteredParameters = null;
+		try {
```
We don't want to keep all the stuff inside of single try section.
If, for example, we will not be able to write Excel file because of file-system problem, application usage (from catch section) will not help us.

artekm replied:
WIll do it when all parts are ready.

olegmiercoles commented on src/main/java/pl/itacademy/schedule/generator/ExcelCreator.java:
```diff
@@ -0,0 +1,103 @@
+	private CellStyle cellStyleLeft;
+	private CellStyle cellStyleCenter;
+
+	public Workbook createHSSFWorkbook(Schedule schedule) {
```
Let's rid of this older format, and keep to newer one - XSSF

artekm replied:
OK

olegmiercoles commented on src/main/java/pl/itacademy/schedule/generator/ExcelCreator.java:
```diff
@@ -0,0 +1,103 @@
+		cell.setCellStyle(style);
+	}
+
+	private void fillCell(Sheet sheet, int row, int column, int value, CellStyle style) {
```
Do we really need both pretty the same methods?
Can we rewrite them to single one for avoid code duplication?

artekm replied:
Perhaps, but don't know how.

olegmiercoles commented on src/main/java/pl/itacademy/schedule/generator/ExcelCreator.java:
```diff
@@ -0,0 +1,103 @@
+		cell.setCellStyle(style);
+	}
+
+	private Cell selectCell(Sheet sheet, int rowNumber, int columnNumber) {
```
Name of the method and what the method really do - are different

artekm replied:
OK?

olegmiercoles commented on src/main/java/pl/itacademy/schedule/generator/ExcelCreator.java:
```diff
@@ -0,0 +1,103 @@
+		return cell;
+	}
+
+	private Row createRow(Sheet sheet, int rowNumber) {
```
Name of the method and what the method really do - are different

artekm replied:
OK?

olegmiercoles commented on src/main/java/pl/itacademy/schedule/generator/ExcelCreator.java:
```diff
@@ -0,0 +1,103 @@
+		return row;
+	}
+
+	private Cell createCell(Row row, int columnNumber) {
```
Name of the method and what the method really do - are different

artekm replied:
OK?

olegmiercoles commented on src/main/java/pl/itacademy/schedule/generator/ExcelCreator.java:
```diff
@@ -0,0 +1,103 @@
+		cellStyleCenter = workbook.createCellStyle();
+		cellStyleCenter.setAlignment(HorizontalAlignment.CENTER);
+	}
+}
```
Also we have some formulas on sheet. 
Check example files: by default they are invisible. There are some option in Excel to show formulas, or try to click on cells.

artekm replied:
Done

olegmiercoles commented on src/main/java/pl/itacademy/schedule/util/PropertiesReader.java:
```diff
@@ -1,42 +1,61 @@
+	private static final String PROPERTIES_FILE_NAME = "application.properties";
+	private static final String SEARCH_LOCATION_A = "./config/";
+	private static final String SEARCH_LOCATION_B = "./";
+	private static final String DEFAULT_EXCEL_NAME = "default";
```
This parameter should be in application.properties

artekm replied:
Done

olegmiercoles commented on src/main/java/pl/itacademy/schedule/util/PropertiesReader.java:
```diff
@@ -1,42 +1,61 @@
+	private static final String SEARCH_LOCATION_A = "./config/";
+	private static final String SEARCH_LOCATION_B = "./";
+	private static final String DEFAULT_EXCEL_NAME = "default";
+	private static final String DEFAULT_EXCEL_EXTENSION = "xls";
```
This parameter should be in application.properties

artekm replied:
Done

olegmiercoles commented on src/main/java/pl/itacademy/schedule/util/PropertiesReader.java:
```diff
@@ -1,42 +1,61 @@
+				try {
+					readDefaultsFromJAR();
+				} catch (IOException e3) {
+					properties.setProperty("excel.defaultName", DEFAULT_EXCEL_NAME);
```
This parameter should be in application.properties

artekm replied:
Done

olegmiercoles commented on src/main/java/pl/itacademy/schedule/util/PropertiesReader.java:
```diff
@@ -1,42 +1,61 @@
+					readDefaultsFromJAR();
+				} catch (IOException e3) {
+					properties.setProperty("excel.defaultName", DEFAULT_EXCEL_NAME);
+					properties.setProperty("excel.defaultExtension", DEFAULT_EXCEL_EXTENSION);
```
This parameter should be in application.properties

olegmiercoles commented on src/main/java/pl/itacademy/schedule/util/PropertiesReader.java:
```diff
@@ -1,42 +1,61 @@
+		input.close();
+	}
+
+	private void readDefaultsFromJAR() throws IOException {
```
Read from JAR looks wrong. Does it really works?
Check example which we had with FileSystem.

olegmiercoles commented on src/test/java/pl/itacademy/schedule/generator/ScheduleGeneratorTest.java:
```diff
@@ -38,4 +39,22 @@ public void generate_generatesScheduleFromParameters() {
     }
 
+    @Test
+    public void generate_generatesScheduleFromParametersWithWarning() {
```
Actually it there are we have no warnings in method.
Probably test name should be different:
generate_requiredHoursCantBeFitIntoDailySchedule_createsLastDayShorer_addsFlagForShorterLastDay
or something like this.

artekm replied:
OK

olegmiercoles commented on src/main/java/pl/itacademy/schedule/ScheduleGeneratorApp.java:
```diff
@@ -1,29 +1,70 @@
-        }
-
-    }
+	public static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("d.MM.uuuu");
```
This should be read as a parameter from application.properties

artekm replied:
Done

olegmiercoles commented on src/main/java/pl/itacademy/schedule/ScheduleGeneratorApp.java:
```diff
@@ -1,29 +1,70 @@
-
-    }
+	public static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("d.MM.uuuu");
+	public static final DateTimeFormatter TIME_FORMATTER = DateTimeFormatter.ofPattern("H:mm");
```
This should be read as a parameter from application.properties

artekm replied:
Done

olegmiercoles reviewed (commented):
So far so good.
You are really surprising me each time more and more.
Still a lot work to do, but Im very happy that each time you do it better!

artekm committed:
Updated after review

-added formulas in Excel
-renamed methods in Excel
-date and time formats are now read from application.properties
-added Unit Test for PropertiesReader
